File: app/src/main/java/com/romankaranchuk/musicplayer/data/Song.java
package com.romankaranchuk.musicplayer.data;

import android.os.Parcel;
import android.os.Parcelable;

import com.romankaranchuk.musicplayer.data.source.MusicDataSource;
import com.romankaranchuk.musicplayer.data.source.MusicRepository;
import com.romankaranchuk.musicplayer.data.source.local.MusicLocalDataSource;
import com.romankaranchuk.musicplayer.utils.MusicUtils;

import java.util.UUID;

/**
 * Created by NotePad.by on 14.10.2016.
 */

public class Song implements Parcelable{
    private String mId;
    private String mName;
    private String mPath;

    private String mImagePath;
    private int mDuration;
    private String mAlbumId;

    private String mLyricsSong;
    private String mYear;
    private String mDate;
    private String mLanguage;

    public Song(String name, String path, String imagePath,
                int duration, String albumId, String lyricsSong,
                String year, String dateModified, String language){
        this(UUID.randomUUID().toString(), name, path,
                imagePath, duration, albumId, lyricsSong, year, dateModified, language);
    }

    public Song(String id, String name, String path,
                String imagePath, int duration, String albumId,
                String lyricsSong, String year, String dateModified, String language){
        mId = id;
        mName = name;
        mPath = path;
        mImagePath = imagePath;
        mDuration = duration;
        mAlbumId = albumId;
        mLyricsSong = lyricsSong;
        mYear = year;
        mDate = dateModified;
        mLanguage = language;
    }

    public String getId(){ return this.mId;}

    public String getPath(){
        return this.mPath;
    }

    public String getLyricsSong(){
        return this.mLyricsSong;
    }

    public String getName(){
        return this.mName;
    }

    public String getTitle(){
        return MusicUtils.extractSongInfo(this.getPath()).title;
    }

    public String getNameArtist(){
        return MusicUtils.extractSongInfo(this.getPath()).artist;
    }

    public String getYear(){
        return mYear;
    }
    public void setName(String mName){
        this.mName = mName;
    }

    public String getImagePath(){
        return this.mImagePath;
    }

    public int getDuration(){
        return this.mDuration;
    }

    public String getAlbumId(){
        return this.mAlbumId;
    }

    public String getDate(){
        return this.mDate;
    }

    public String getLanguage(){
        return this.mLanguage;
    }
    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(this.mId);
        parcel.writeString(this.mName);
        parcel.writeString(this.mPath);
        parcel.writeString(this.mImagePath);
        parcel.writeString(this.mLyricsSong);
        parcel.writeInt(this.mDuration);
        parcel.writeString(this.mYear);
        parcel.writeString(this.mAlbumId);
        parcel.writeString(this.mDate);
        parcel.writeString(this.mLanguage);
    }

    private Song(Parcel parcel){
        mId = parcel.readString();
        mName = parcel.readString();
        mPath = parcel.readString();
        mImagePath = parcel.readString();
        mLyricsSong = parcel.readString();
        mDuration = parcel.readInt();
        mYear = parcel.readString();
        mAlbumId = parcel.readString();
        mDate = parcel.readString();
        mLanguage = parcel.readString();
    }


    public static final Parcelable.Creator CREATOR = new Parcelable.Creator(){
        public Song createFromParcel(Parcel in){
            return new Song(in);
        }

        public Song[] newArray(int size){
            return new Song[size];
        }
    };
}

File: app/src/main/java/com/romankaranchuk/musicplayer/data/Song.kt
package com.romankaranchuk.musicplayer.data

import android.os.Parcelable
import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.romankaranchuk.musicplayer.utils.MusicUtils
import kotlinx.android.parcel.Parcelize
import java.util.*

@Entity(tableName = "songs")
@Parcelize
data class Song (
    @NonNull
    @PrimaryKey
    @ColumnInfo(name = "song_id")
    var id: String,

    @ColumnInfo(name = "song_name")
    var name: String?,

    @ColumnInfo(name = "song_path")
    var path: String?,

    @ColumnInfo(name = "song_image")
    var imagePath: String?,

    @ColumnInfo(name = "song_duration")
    var duration: Int?,

    @ColumnInfo(name = "album_id")
    var albumId: String,

    @ColumnInfo(name = "song_lyrics")
    var lyricsSong: String?,

    @ColumnInfo(name = "song_year")
    var year: String?,

    @ColumnInfo(name = "song_date")
    var date: String?,

    @ColumnInfo(name = "song_language")
    var language: String?,
) : Parcelable {

    constructor(
        name: String?,
        path: String?,
        imagePath: String?,
        duration: Int?,
        albumId: String,
        lyricsSong: String?,
        year: String?,
        dateModified: String?,
        language: String?
    ) : this(
        id = UUID.randomUUID().toString(),
        name = name,
        path = path,
        imagePath = imagePath,
        duration = duration,
        albumId = albumId,
        lyricsSong = lyricsSong,
        year = year,
        date = dateModified,
        language = language
    )

    val title: String
        get() = MusicUtils.extractSongInfo(path).title
    val nameArtist: String
        get() = MusicUtils.extractSongInfo(path).artist
}